### PR TITLE
solution

### PR DESCRIPTION
**Pull request by shortSparrow**

### CODE REVIEW

File: src/calculateAverage.js [hunk 1]
```diff
@@ -32,12 +32,40 @@
...
+    switch (typeof options) {
+      case 'string':
+        return undefined;
+      case 'object':
+        return Obj.call(this);
+      default:
+        return this.reduce((a, b) => a + b) / this.length;
+    }
+
+    function Obj() {
```
Comment (irenhh) [inline src/calculateAverage.js]: function name starts with lowercase letter

File: src/calculateAverage.js [hunk 2]
```diff
@@ -32,12 +32,40 @@
...
+      if (options === null || (isNaN(options) && typeof options !== 'object')) {
+        return undefined;
+      }
+
+      const key = Object.values(options)[0];
+      let iteribleValues = null;
```
Comment (irenhh) [inline src/calculateAverage.js]: iteribleValues?

File: src/calculateAverage.js [hunk 3]
```diff
@@ -32,12 +32,40 @@
...
+
+      if (typeof key === 'function') {
+        iteribleValues = this.map(key);
+      } else {
+        iteribleValues = this
+          .map(obj => {
+            return obj[key];
```
Comment (irenhh) [inline src/calculateAverage.js]: no need to write `return` explicitly

**Review by irenhh — CHANGES_REQUESTED** [review]

File: src/calculateAverage.js [hunk 4]
```diff
@@ -37,34 +37,32 @@ function applyCalculateAverage() {
...
         return undefined;
       case 'object':
-        return Obj.call(this);
+        return objCase.call(this);
       default:
         return this.reduce((a, b) => a + b) / this.length;
     }
 
-    function Obj() {
+    function objCase() {
```
Comment (irenhh) [inline src/calculateAverage.js]: why `objCase`? what does this function do?

File: src/calculateAverage.js [hunk 5]
```diff
@@ -37,34 +37,32 @@ function applyCalculateAverage() {
...
       if (options === null || (isNaN(options) && typeof options !== 'object')) {
         return undefined;
       }
 
       const key = Object.values(options)[0];
-      let iteribleValues = null;
+      let values = null;
```
Comment (irenhh) [inline src/calculateAverage.js]: values of what?
choose more specific name

**Review by irenhh — CHANGES_REQUESTED** [review]

File: src/calculateAverage.js (hunk 4 above)
Comment (nizkiyVorobey) [reply src/calculateAverage.js]: работать если case = object

File: src/calculateAverage.js [hunk 6]
```diff
@@ -37,32 +37,32 @@ function applyCalculateAverage() {
       case 'string':
         return undefined;
       case 'object':
-        return objCase.call(this);
+        return AverageIntoObj.call(this);
```
Comment (irenhh) [inline src/calculateAverage.js]: did you forget how to name variables in js?

Comment (irenhh) [reply src/calculateAverage.js]: average what?

**Review by irenhh — CHANGES_REQUESTED** [review]

**Pull request closed without merging**